### PR TITLE
Added AMQP Transport

### PR DESCRIPTION
**Pull request by zhdanovartur**

### CODE REVIEW

**Comment by bzikarsky** [comment]
Can you provide tests please? I'd rather have every feature fully tested. 


Also: This implements #1.

**Comment by zhdanovartur** [comment]
Done

**Comment by zhdanovartur** [comment]
Done, but Travis has some error with composer transport in php 5.3.
Try rebuild it later.

File: src/Gelf/Transport/AmqpTransport.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    public function send(Message $message)
+    {
+        $rawMessage = $this->getMessageEncoder()->encode($message);
+        $this->exchange->publish(
+            $rawMessage,
+            $this->queue->getName(),
+            0,
+            array(
+                'delivery_mode' => 2,
```
Comment (bzikarsky) [inline src/Gelf/Transport/AmqpTransport.php]: This only makes sense with `$queue->setFlags(AMQP_DURABLE);`, doesn't it? 
Would it be better to only add `delivery_mode = 2` if `($this->queue->getFlags() & AMQP_DURABLE) > 0`?

**Comment by bzikarsky** [comment]
I rescheduled the failed 5.3.0 build. It went through this time - it was an error on packagist's side.

**Comment by bzikarsky** [comment]
Thanks - One final step and I am ready to merge; can you squash the commits into a single feature commit, which includes "resolves #1" in its commit message? If you need help with squashing, check out: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes

**Comment by zhdanovartur** [comment]
It's ok?

**Comment by bzikarsky** [comment]
Try this:
    
    git remote add bz https://github.com/bzikarsky/gelf-php.git
    git fetch bz
    git merge-base master bz/master
    # this returns the hash of the merge-base-commit
    git rebase -i #hash 

In the opened editor, tag the first commit as "reword", all other lines as "squash". Then rewrite the the commit message and in the end do a force push with

    git push --force origin master

**Comment by zhdanovartur** [comment]
Rebuild please, Travis has same error with composer, but for PHP 7

**Pull request merged**